plugins/modules/ndo_service_device_cluster.py

```diff
@@ -0,0 +1,641 @@
+                    },
+                    mutually_exclusive=[
+                        ["template", "template_id"],
+                    ],
+                ),
+                ipsla_monitoring_policy_uuid=dict(type="str"),
+                qos_policy=dict(
+                    type="dict",
+                    options=dict(
+                        name=dict(type="str"),
+                        template=dict(type="str"),
+                        template_id=dict(type="str"),
+                    ),
+                    required_by={
+                        "template": "name",
+                        "template_id": "name",
+                    },
+                    mutually_exclusive=[
+                        ["template", "template_id"],
+                    ],
+                ),
+                qos_policy_uuid=dict(type="str"),
+                preferred_group=dict(type="bool"),
+                rewrite_source_mac=dict(type="bool"),
+                anycast=dict(type="bool"),
+                config_static_mac=dict(type="bool"),
+                is_backup_redirect_ip=dict(type="bool"),
+                load_balance_hashing=dict(type="str", choices=["sourceDestinationAndProtocol", "sourceIP", "destinationIP"]),
+                pod_aware_redirection=dict(type="bool"),
+                resilient_hashing=dict(type="bool"),
+                tag_based_sorting=dict(type="bool"),
+                min_threshold=dict(type="int"),
+                max_threshold=dict(type="int"),
+                threshold_down_action=dict(type="str", choices=["permit", "deny", "bypass"]),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["name", "uuid"], True],
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["device_mode", "device_type", "interface_properties"]],
+        ],
+        required_one_of=[["template", "template_id"]],
+        mutually_exclusive=[
+            ["template", "template_id"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    device_type = module.params.get("device_type")
+    device_mode = module.params.get("device_mode")
+    connectivity_mode = module.params.get("connectivity_mode")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+    device_path = None
+
+    mso_template = MSOTemplate(mso, "service_device", template, template_id)
+    mso_template.validate_template("serviceDevice")
+    references = {"bd": "bdRef", "l3out": "externalEpgRef"}
+    connectivity_mode = {1: "oneArm", 2: "twoArm"}
```

Comment:
why define as var when only used in single condition? perhaps better readable if declared in the condition or rewrite the else part `"advanced" if len(interface_properties) >= 3 else  ("oneArm" if 1 else "twoArm")`